compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -463,6 +463,19 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                                 ),
                                 root_obligation,
                             )
+                        } else if matches!(
+                            obligation.cause.code(),
+                            ObligationCauseCode::ImplDerivedObligation(c) if tcx.has_attr(c.impl_or_alias_def_id, sym::do_not_recommend)
+                        ) {
+                            let (code, base) = obligation.cause.code().peel_derives_with_predicate();
+                            if let Some(base) = base {
+                                let code = code.clone();
+                                obligation.cause.map_code(|_| code);
+                                // FIXME: somehow we need to overwrite obligation.predicate here
```

Comment:
I believe that I need to change something with the obligation here to set the updated predicate so that the error label stops mentioning `Expression` as well. I just couldn't figure out how to do that yet. Help would be welcome here.

Reply:
That's resolved now